clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/ClusteredStateRepository.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.client.internal.service;
+
+import org.ehcache.clustered.client.service.ClusteringService;
+import org.ehcache.spi.persistence.StateRepository;
+
+import java.io.Serializable;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * ClusteredStateRepository
+ */
+class ClusteredStateRepository implements StateRepository {
+
+  private final ClusteringService.ClusteredCacheIdentifier clusterCacheIdentifier;
+  private final String composedId;
+  private final DefaultClusteringService defaultClusteringService;
+
+  ClusteredStateRepository(ClusteringService.ClusteredCacheIdentifier clusterCacheIdentifier, String id, DefaultClusteringService defaultClusteringService) {
+    this.clusterCacheIdentifier = clusterCacheIdentifier;
+    this.composedId = clusterCacheIdentifier.getId() + "-" + id;
```

Comment:
Because I did not want the `DefaultClusteringService` to have anything to do with the naming scheme and have it instead resolved at this level completely. The fact that I end up using the `ClusteredCacheIdentifier.getId()` is an impl detail.